java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -276,21 +276,26 @@ private void buildCall(Tree element, UCFGBuilder.BlockBuilder blockBuilder, Iden
       buildPlusAssignmentInvocation(blockBuilder, idGenerator, assignmentExpressionTree);
     } else if (element.is(ARRAY_ACCESS_EXPRESSION) && !element.parent().is(PLUS_ASSIGNMENT, ASSIGNMENT)) {
       // PLUS_ASSIGNMENT and ASSIGNMENT might imply an array set, otherwise an array access is always a get
-      Expression.Variable getValue = variableWithId(idGenerator.newId());
-      Expression array = idGenerator.lookupExpressionFor(((ArrayAccessExpressionTree)element).expression());
-      blockBuilder.assignTo(getValue, arrayGet(array), location(element));
-      idGenerator.varForExpression(element, getValue.id());
+      ArrayAccessExpressionTree arrayAccessExpressionTree = (ArrayAccessExpressionTree) element;
+      if (isObject(arrayAccessExpressionTree.symbolType())) {
```

Comment:
I'm not 100% sure that this will not always return true. All arrays are not primitives. I believe that you should look at the type of the `ArrayType.elementType()` (which can be another array).

Reply:
In the current implementation, the type of an `ArrayAccessExpressionTree` is the type of the element that is accessed. It is false for the tests with `int` and `byte`

Reply:
So if for instance its a 2-dimensional integer array `int[][] myArray`, the type of `myArray[0]` is going to be `int[]`, translated into `$Array`, which we still don't want.